Update Nextjs quickstart (#806)
clerk/clerk-docs: docs/quickstarts/nextjs.mdx

Reviewer: There's a line that says "Select your preferred router below" and I think we should change it to "In the following example, select your preferred router"

I can't make a suggestion because its not a line edited in this PR.

### Set your environment variables

> 1. If you're signed into Clerk, the `.env.local` snippet below will contain your keys. Otherwise, get your API keys from your [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=api-keys).

Reviewer: If the users aren't signed in to their dashboard, that link will not take them to the desired page. So we need to have copy that tells users which page the API keys lives on. I've grabbed the language that we use on other pages, and inserted it into the suggestion.

Also, iirc, your previous suggestion was to refrain from using "below" because it relies on visual positioning, which may not be universally understood by all users. Blind or visually impaired users who rely on screen readers, for example, may not perceive the spatial relationship implied by "below." I've suggested we use "following" instead !

```suggestion
1. If you are signed into Clerk, the following `.env.local` snippet will contain your keys. Otherwise, you can find your keys in the Clerk Dashboard on the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page.

```

Now that Clerk is installed and mounted in your application, you can decide which pages are public and which should require authentication to access:

1. Create a `middleware.ts` file at the root of your project, or in your `src/` directory if you have one.

> 1. In your `middleware.ts`, export Clerk's [`authMiddleware()`](https://clerk.com/docs/references/nextjs/auth-middleware) helper. This helper enables authentication and blocks access for signed out visitors on routes that your middleware runs on.

Reviewer: ```suggestion
2. In your `middleware.ts`, export Clerk's [`authMiddleware()`](https://clerk.com/docs/references/nextjs/auth-middleware) helper. This helper enables authentication and blocks access for signed out visitors on routes that your middleware runs on.

```

> 1. In your `middleware.ts`, export Clerk's [`authMiddleware()`](https://clerk.com/docs/references/nextjs/auth-middleware) helper. This helper enables authentication and blocks access for signed out visitors on routes that your middleware runs on.

Reviewer 2: ```suggestion
1. In your `middleware.ts` file, export Clerk's [`authMiddleware()`](https://clerk.com/docs/references/nextjs/auth-middleware) helper. This helper enables authentication and blocks access for signed out visitors on routes that your middleware runs on.

```

    Next.js middleware only runs on routes specified with [`matcher`](https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher). Add the following code to your `middleware.ts` to protect your routes:

    ```tsx filename="middleware.ts"
    import { authMiddleware } from "@clerk/nextjs";

    // See https://clerk.com/docs/references/nextjs/auth-middleware
    // for more information about configuring your Middleware

    export default authMiddleware({
      // Allow signed out users to access the specified routes:

>       // publicRoutes: ['/anyone-can-visit-this-route'],
>       // Prevent the specified routes from accessing
>       // authentication information:
>       // ignoredRoutes: ['/no-auth-in-this-route'],

Reviewer: Did you mean to comment ALL of this out?

```suggestion
      publicRoutes: ['/anyone-can-visit-this-route'],
      // Prevent the specified routes from accessing
      // authentication information:
      ignoredRoutes: ['/no-auth-in-this-route'],

```

Author: Yes, Colin requested "Can we comment out the scaffolded ignoredRoutes and publicRoutes" [here](https://clerkinc.slack.com/archives/C01QFRQNHSN/p1710219197063989)

    });

    export const config = {
      matcher: [

>         // 1. Exclude files with a "." followed by an extension, which are typically static files.
>         // 2. Exclude files in the _next directory, which are Next.js internals.

Reviewer: The numbers here are telling me that they correspond to a line 1 and a line 2 below them, but they both only correspond to one single line of code. Not sure if the numbers are needed here.

```suggestion
        // Exclude files with a "." followed by an extension, which are typically static files.
        // Exclude files in the _next directory, which are Next.js internals.

```

Now that Clerk is installed and mounted in your application, you can decide which pages are public and which should require authentication to access:

1. Create a `middleware.ts` file at the root of your project, or in your `src/` directory if you have one.

> 1. In your `middleware.ts` file, export Clerk's [`authMiddleware()`](https://clerk.com/docs/references/nextjs/auth-middleware) helper. This helper enables authentication and blocks access for signed out visitors on routes that your middleware runs on.

Reviewer: ```suggestion
2. In your `middleware.ts` file, export Clerk's [`authMiddleware()`](https://clerk.com/docs/references/nextjs/auth-middleware) helper. This helper enables authentication and blocks access for signed out visitors on routes that your middleware runs on.

```

    Next.js middleware only runs on routes specified with [`matcher`](https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher). Add the following code to your `middleware.ts` to protect your routes:

    ```tsx filename="middleware.ts"
    import { authMiddleware } from "@clerk/nextjs";

    // See https://clerk.com/docs/references/nextjs/auth-middleware
    // for more information about configuring your Middleware

    export default authMiddleware({
      // Allow signed out users to access the specified routes:
      // publicRoutes: ['/anyone-can-visit-this-route'],
      // Prevent the specified routes from accessing
      // authentication information:
      // ignoredRoutes: ['/no-auth-in-this-route'],
    });

    export const config = {
      matcher: [
        // Exclude files with a "." followed by an extension, which are typically static files.
        // Exclude files in the _next directory, which are Next.js internals.

> 

Reviewer: ```suggestion
```

Then, visit your app's homepage at [`localhost:3000`](http://localhost:3000) to see the `<UserButton />` in action. It should show your avatar from the account you signed in with.

### Sign out

Finally, use the `<UserButton />` by selecting your avatar, then selecting **Sign out**.

</Steps>

## Next steps

To see an example application that demonstrates user authentication, and session and organization management, explore one of Clerk's Next.js demo repos:

> - [Clerk + Next.js App Router Demo](https://github.com/clerk/clerk-nextjs-demo-app-router)

Reviewer: I don't think we should remove these fully - users use these as demonstrative repositories of our features, and for their code. We can instead do what we did in the Core 2 docs: https://beta.clerk.com/docs/quickstarts/nextjs

<img width="755" alt="Screenshot 2024-03-13 at 12 52 16" src="https://github.com/clerk/clerk-docs/assets/98043211/a7f13358-63ac-4c5a-b248-66a9c9b78f1a">
